src/Cache/Translation_Cache.php

```diff
@@ -0,0 +1,189 @@
+<?php declare(strict_types=1);
+
+namespace Tribe\Weglot\Cache;
+
+use Tribe\Libs\Cache\Cache;
+
+class Translation_Cache {
+
+	protected Cache $cache;
+
+	/**
+	 * In memory cache of the language_code => 'content' for
+	 * the current post.
+	 *
+	 * @var array<int, array<string, string>>
+	 */
+	protected array $translation_cache = [];
+
+	/**
+	 * In memory cache of this post's reference keys.
+	 *
+	 * @var array<string, string>|null
+	 */
+	protected ?array $reference_cache = null;
+
+	/**
+	 * Enable in-memory caching.
+	 */
+	protected bool $use_memory_cache;
+
+	public function __construct( Cache $cache, bool $use_memory_cache = true ) {
+		$this->cache            = $cache;
+		$this->use_memory_cache = $use_memory_cache;
+	}
+
+	/**
+	 * Get a translation cache key for a post.
+	 *
+	 * @param int $post_id
+	 *
+	 * @return string
+	 */
+	public function get_cache_key( int $post_id ): string {
+		return sprintf( 'tribe_translated_%d', $post_id );
+	}
+
+	/**
+	 * Returns all cached content translations for a post.
+	 *
+	 * @param int $post_id
+	 *
+	 * @return array<string, string>
+	 */
+	public function get( int $post_id ): array {
+		if ( $this->use_memory_cache && ! empty( $this->translation_cache[ $post_id ] ) ) {
+			return $this->translation_cache[ $post_id ];
+		}
+
+		$references = $this->get_reference_map( $post_id );
+
+		$translations = [];
+
+		foreach ( $references as $code => $key ) {
+			$translations[ $code ] = $this->get_translated_content( $key );
+		}
+
+		$this->translation_cache[ $post_id ] = $translations;
+
+		return $translations;
+	}
+
+	/**
+	 * Return a specific language translation for a post.
+	 *
+	 * @param string $language_code The two letter language code, e.g. `de`
+	 * @param int    $post_id
+	 *
+	 * @return string
+	 */
+	public function get_by_language( string $language_code, int $post_id ): string {
+		$references = $this->get_reference_map( $post_id );
+
+		$key = $references[ $language_code ] ?? '';
+
+		if ( ! $key ) {
+			return '';
+		}
+
+		return $this->get_translated_content( $key );
+	}
+
+	/**
+	 * Appends a language translation to an existing post's translation
+	 * cache array.
+	 *
+	 * @param mixed  $content
+	 * @param string $language_code
+	 * @param int    $post_id
+	 * @param int    $expire
+	 *
+	 * @return bool
+	 */
+	public function set( $content, string $language_code, int $post_id, int $expire = 0 ): bool {
+		$reference_key = $this->build_reference_key( $post_id, $language_code );
+		$reference_map = array_filter( array_merge( $this->get_reference_map( $post_id ), [
+			$language_code => $reference_key,
+		] ) );
+
+		// Store the translated content in its own unique key.
+		if ( ! $this->cache->set( $reference_key, $content, 'tribe', $expire ) ) {
```

Comment:
The problem is group clearing doesn't actually work. The way it's supposed to work is the Cache class should be changing the reference keys in the options table so that it no longer looks up the same data, but I think that check itself remains cached in memcached. I think it's a bug in tribe-libs, but I wasn't sure so I just refactored to make sure we actually delete the data from the cache instead.

Reply:
Also `tribe` is the default group in the Cache class, but if you pass an empty string it'll use that, so that's really the only reason I had to include it here (because all the `$this->cache->get()` are all using the default `tribe` group but not specifically defined).